ui/src/components/addItemStep3.js

```diff
@@ -0,0 +1,229 @@
+            country:props.data.address?props.data.address.country:'',
+            zip:props.data.address?props.data.address.zip:'',
+            phone:props.data.address?props.data.address.phone:'',
+            msg:{},
+            countriesAutocomplete:Object.keys(citiesAndCountries).map(x=>({id:x,label:x})),
+            citiesAutocomplete:[]
+        }
+        console.log(props);
+        console.log(this.state);
+    }
+        
+    componentDidMount=()=>{
+        this.addressField=document.querySelector('#address')
+        const loader=new Loader(MAPS_API_KEY,{
+            version: "weekly",
+            libraries: ["places"]
+        });
+
+        loader.load().then((google)=>{
+            let autocomplete = new google.maps.places.Autocomplete(this.addressField, {
+                fields: ["address_components", "geometry"],
+                types: ["address"],
+            });
+
+            const fillData=()=>{
+                const place = autocomplete.getPlace();
+                this.state.address="";
+                for (const component of place.address_components){
+                    const compType=component.types[0];
+                    switch (compType) {
+                        case "street_number": {
+                            this.setState({['address']:`${this.state.address} ${component.long_name}`});
+                            break;
+                        }
+                        case "route": {
+                            this.setState({['address']:`${component.short_name} ${this.state.address}`});
+                            break;
+                        }
+                        case "postal_code": {
+                            this.setState({['zip']:component.long_name});
+                            break;
+                        }
+                        case "locality":{
+                            this.setState({['city']:component.long_name});
+                            break;
+                        }
+                        case "country":{
+                            this.setState({['country']:component.long_name});
+                            break;
+                        }
+                    }
+                }
+            }
+
+            autocomplete.addListener("place_changed", fillData);
+        });
+    }
+
+    onChange=(e)=>{
+        this.setState({[e.target.name]:e.target.value});
+    }
+
+    onNext=()=>{
+        let valid=true;
+        let msg={};
+        if(!this.state.address||!this.state.address.length>0){
+            valid=false;
+            msg.address="Address name can't be empty";
+        }
+        if(!this.state.city||!this.state.city.length>0){
+            valid=false;
+            msg.city="City name can't be empty";
+        }
+        if(!this.state.zip||!this.state.zip.length>0){
+            valid=false;
+            msg.zip="ZIP code can't be empty";
+        }
+        if(!this.state.country||!this.state.country.length>0){
```

Comment:
it wil be enough !this.state.country

empty string is evalueated ad falsy value